020-Getting-started/070-nextjs.mdx

```diff
@@ -50,185 +46,621 @@ apiKey=YOUR_API_KEY_HERE
 
-```tsx {1-3}
-import { getXataClient } from '../xata';
-// We'll assign the client to xata to make it easier to call
 const xata = getXataClient();
 
-export default function PostsListPage() {
+export default async function Home() {
+  const posts = await xata.db.Posts.getAll();
+
   return (
-    <>
-      <h1>My xata posts</h1>
-    </>
+    <main className="flex flex-wrap justify-center content-start p-8 lg:p-24 min-h-screen">
+      <div className="z-10 h-50 w-full max-w-5xl items-center justify-between text-xl lg:flex">
+        <p className="fixed left-0 top-0 flex w-full justify-center pb-6 pt-8 lg:static lg:w-auto bg-gradient-to-b from-white via-white via-65% dark:from-black dark:via-black lg:bg-none">
+          <a href="/">Get started with Xata and Next.js</a>
+        </p>
+        <div className="fixed bottom-0 left-0 flex w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+          <a href="https://xata.io" className="w-20">
+            <img src="https://raw.githubusercontent.com/leggetter/xata-examples/5b0715c9ba36556400b6121bd224203cb89f7741/docs/app_logo.svg" />
+          </a>
+        </div>
+      </div>
+
+      <div className="w-full max-w-5xl mt-16">
+        {posts.length === 0 && <p>No blog posts found</p>}
+        {posts.map((record) => (
+          <div key={record.id} className="mb-16">
+            <p className="text-xs mb-2 text-purple-950 dark:text-purple-200">
+              {record.pubDate?.toDateString()}
+            </p>
+            <h2 className="text-2xl mb-2">
+              <a href={`posts/${record.slug}`}>{record.title}</a>
+            </h2>
+            <p className="text-purple-950 dark:text-purple-200 mb-5">
+              {record.description}
+            </p>
+            <a
+              href={`posts/${record.slug}`}
+              className="px-4 py-2 font-semibold text-sm bg-purple-700 text-white rounded-lg shadow-sm w-fit"
+            >
+              Read more &rarr;
+            </a>
+          </div>
+        ))}
+      </div>
+    </main>
   );
 }
 ```
 
-To fix the error we'll need to utilize the Xata client in Next.js's server-side mode. This can be done with the `getServerSideProps` function that Next.js provides.
+Let's break down what's happening in the code above.
+
+First, we import the `getXataClient` function and assign the result of calling that function to a variable named `xata`:
+
+```tsx
+import { getXataClient } from '@/xata';
 
-```tsx {4-13}
-import { getXataClient } from '../xata';
 const xata = getXataClient();
+```
+
+Then, in the `Home` function, use the `xata` client instance to get all the posts stored in the database. This is
+achieved via the auto-generated `Posts` property which exposes a number of helper functions. In this case, we use the
+[`getAll`](https://xata.io/docs/typescript-client/get#the-typescript-sdk-functions-for-querying) function to get
```

Comment:
Nit: It doesn't really make a difference in this case, but `getAll()` is somewhat dangerous because it really gets all by doing multiple round trips. Consider changing to `getMany()`.

Reply:
Should be getPaginated instead

Reply:
This box clarification looks to me. Thanks!